React/src/App.jsx

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useState } from 'react';
+import Scheduler from 'devextreme-react/scheduler';
+import { Popup, Position, ToolbarItem } from 'devextreme-react/popup';
+import { isOverlapRecurrentAppointment } from './utils/isOverlapRecurrentAppointment';
+import { defaultData } from './data.js';
+import 'devextreme/dist/css/dx.common.css';
+import 'devextreme/dist/css/dx.material.blue.light.css';
+import './App.css';
+
+const App = () => {
+    const [popupVisible, setPopupVisible] = useState(false);
+    const [data] = useState(defaultData);
+
+    const hideInfo = () => {
+        setPopupVisible(false);
+    }
+
+    const closeButtonOptions = {
+        text: 'Close',
+        onClick: hideInfo,
+    };
+
+    const handleAppointmentActions = (
+        event,
+        recurrentAppointments,
+        newAppointment) => {
+        for (const recurrentAppointment of recurrentAppointments) {
+            const isOverlap = isOverlapRecurrentAppointment(
+                event,
+                recurrentAppointment,
+                newAppointment);
+            if (isOverlap) {
+                cancelAppointmentAdding(event);
+            }
+        }
+    }
```

Comment:
Let's add ```useCallback``` and get rid of ```cancelAppointmentAdding``` method that used only here:

```suggestion
    const handleAppointmentActions = useCallback((
        event,
        recurrentAppointments,
        newAppointment) => {
        for (const recurrentAppointment of recurrentAppointments) {
            const isOverlap = isOverlapRecurrentAppointment(
                event,
                recurrentAppointment,
                newAppointment);
            if (isOverlap) {
                event.cancel = true;
                setPopupVisible(true)
            }
        }
    }, []);
```